src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -0,0 +1,134 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.CPU;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.NETWORK;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModifyQueueCapacityAction implements Action {
+
+  public static final String NAME = "modify_queue_capacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
+
+  private int currentCapacity;
+  private int desiredCapacity;
+  private ResourceEnum threadPool;
+  private NodeKey esNode;
+
+  private Map<ResourceEnum, Integer> lowerBound = new HashMap<>();
+  private Map<ResourceEnum, Integer> upperBound = new HashMap<>();
+
+  public ModifyQueueCapacityAction(NodeKey esNode, ResourceEnum threadPool, int currentCapacity, boolean increase) {
+    setBounds();
+    int STEP_SIZE = 50;
+    this.esNode = esNode;
+    this.threadPool = threadPool;
+    this.currentCapacity = currentCapacity;
+    int desiredCapacity = increase ? currentCapacity + STEP_SIZE : currentCapacity - STEP_SIZE;
+    setDesiredCapacity(desiredCapacity);
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public boolean isActionable() {
+    return desiredCapacity != currentCapacity;
+  }
+
+  @Override
+  public int coolOffPeriodInSeconds() {
+    return COOL_OFF_PERIOD_IN_SECONDS;
+  }
+
+  @Override
+  public List<NodeKey> impactedNodes() {
+    return Collections.singletonList(esNode);
+  }
+
+  @Override
+  public Map<NodeKey, ImpactVector> impact() {
+    ImpactVector impactVector = new ImpactVector();
+    if (desiredCapacity > currentCapacity) {
+      impactVector.increasesPressure(HEAP, CPU, NETWORK);
+    } else if (desiredCapacity < currentCapacity) {
+      impactVector.decreasesPressure(HEAP, CPU, NETWORK);
+    }
+    return Collections.singletonMap(esNode, impactVector);
+  }
+
+  @Override
+  public void execute() {
+    // Making this a no-op for now
+    // TODO: Modify based on downstream agent API calls
+    assert true;
+  }
+
+  @Override
+  public String summary() {
+    if (!isActionable()) {
```

Comment:
As stepsize is a positive integer, the creation of an action means that the desired size `!=` currentCapacity as here:
`desiredCapacity = increase ? currentCapacity + STEP_SIZE : currentCapacity - STEP_SIZE;`

What are the cases when it is not actionable ?